hudi-spark/src/main/java/org/apache/hudi/internal/HoodieDataSourceInternalWriter.java

```diff
@@ -102,7 +104,9 @@ public void commit(WriterCommitMessage[] messages) {
             .flatMap(m -> m.getWriteStatuses().stream().map(m2 -> m2.getStat())).collect(Collectors.toList());
 
     try {
-      writeClient.commitStats(instantTime, writeStatList, Option.empty());
+      writeClient.commitStats(instantTime, writeStatList, Option.empty(),
```

Comment:
better approach for these situations generally is to introduce a `commitStats(.)` that does not take the last argument and deal with it internally inside WriteClient. 

This way the code will remain more readable, without having to reference replace stats in bulk_insert, which have nothing to do with each other. hope that makes sense

Reply:
Sure. Added new method